Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -182,101 +209,99 @@ public void ComeToStop()
+				foreach ( Collider col in colliders )
 				{
-					Collider[] cols = Physics.OverlapSphere( transform.position, climbCheckRadius, climbLayer );
-					if ( cols.Length > 0 )
+					float distance =  ( col.transform.position - transform.position ).sqrMagnitude;
+					if ( distance < shortestDistance )
 					{
-						Collider nearestCol = cols[0];
-						foreach ( Collider col in cols )
-						{
-							if ( ( col.transform.position - transform.position ).sqrMagnitude < ( nearestCol.transform.position - transform.position ).sqrMagnitude )
-							{
-								nearestCol = col;
-							}
-						}
-
-						StartClimbing( nearestCol );
-					}
-					else
-					{
-						StopClimbing();
+						nearestCol = col;
+						shortestDistance = distance;
 					}
 				}
 
-				climbCheckTimer = 0f;
+				StartClimbing( nearestCol );
+				climbing = true;
 			}
+			else
+			{
+				StopClimbing();
+				climbing = false;
+			}
+
+			climbCheckTimer = 0f;
 		}
-		else if ( climbSurface )
-		{
-			StopClimbing();
-		}
+		//else
+		//{
+		//	climbing = true;
+		//}
 		
-		climbCheckTimer += Time.fixedDeltaTime;
+		return climbing;
 	}
 
-	public void ClimbSurface()
+	public void ClimbSurface( Vector3 movement )
 	{
-		if(climbSurface)
+		DebugUtils.Assert( climbSurface, "Cannot climb, surface is null" );
+
+		if ( climbTag )
 		{
-			ClimbableTag ct = climbSurface.GetComponent<ClimbableTag>();
-			if(ct)
-			{
-				rigidbody.useGravity = false;
-				currStoppingPower = stoppingSpeed;
+			DebugUtils.Assert( climbTag.xMovement || climbTag.yMovement );
 
-				Vector3 surfaceRelativeInput = climbSurface.InverseTransformDirection( inputVec );
-				surfaceRelativeInput = new Vector3( surfaceRelativeInput.x, 
-				                                    surfaceRelativeInput.z, 
-				                                    surfaceRelativeInput.y );
+			currStoppingPower = stoppingSpeed;
 
-				if ( !ct.XMovement ) surfaceRelativeInput.x = 0f;
-				if ( !ct.YMovement ) surfaceRelativeInput.y = 0f;
+			Vector3 surfaceRelativeInput =
+			    climbSurfaceRight * ( climbTag.xMovement ? movement.x : 0.0f ) +
+			    Vector3.up * ( climbTag.yMovement ? movement.z : 0.0f );
 
-				Vector3 surfaceHoldVec = climbSurface.forward * surfaceHoldForce;
-				moveVec = climbSurface.rotation * surfaceRelativeInput * climbMoveSpeed * moveSpeedMod;
-				moveVec += surfaceHoldVec;
+			Debug.DrawRay( transform.position, surfaceRelativeInput * 10.0f );
 
-				lastVelocity = moveVec;
-				rigidbody.velocity = moveVec;
+			moveVec = surfaceRelativeInput * climbMoveSpeed * moveSpeedMod;
 
-				model.rotation = Quaternion.Lerp( model.rotation, 
-				                                  Quaternion.LookRotation( climbSurface.forward, Vector3.up ), 
-				                                  Time.deltaTime * modelTurnSpeed );
-			}
-			else
-			{
-				Debug.LogError("Cannot climb, surface isn't tagged. This is probably a problem in ClimbCheck");
-			}
+			transform.Translate( moveVec * Time.deltaTime, Space.World );
```

Comment:
This makes me so unhappy, but only because I hate transform.Translate irrationally. I'd totally set velocity here.